### DISCUSSION TITLE
Error load dynamic treesitter library

### DISCUSSION THREAD

**Discussion by yonchicy** [helix-editor/helix#8190] [General]
Sorry if this is the wrong place for this question. I have tried searching all over to no avail.
I want to use helix on an offline server with a lower version glic. So i built from source with following instructions
```shell
RUSTFLAGS="-C target-feature=-crt-static"
cargo build --target x86_64-unknown-linux-musl --release
```
Then I used `ldd` to confirm the that both `hx` binary and treesitter `so` files were statically linked.
Then I copyed `hx` binary and `runtime` directory to my server ,and `runtime` files were copyed  in `~/.config/helix` .
 I run `hx -v a.c` and LSP worked but there was no code highlights.
Log shows follows:
>2023-09-05T12:49:53.617 helix_core::syntax [ERROR] Failed to load tree-sitter parser for language "c": Error opening dynamic library "/home/oahu/.config/helix/runtime/grammars/c.so"

Is  there any thing else i can try?

**Comment by pascalkuthe** [comment]
Helix loads the tree sitter grammar as dynamic libraries. Static linking musl is not possible in this case. You can build with musk but then you will have to dynamic link it (and therefore musl also needs to be present on the server).

It's probably a better idea to spin up a centos container and build on there to target an older glibc version.

**Reply by Ktoks** [reply]
I will try this out and see what happens - I'm running into the same problem on my work machines (rhel 8).
